### ISSUE TITLE
License

### ISSUE THREAD

**Issue by J-Endres** [mzaiss/MRTwin_pulseq#6]
The build tool / dependency used to build the pre-pass as a python module is licensed under Apache-2.0. This license requires a copy of the license itself, as well as a copyright notice.
All other pre-pass dependencies are MIT, which may or may not require the same.

We don't need to include any copyright notices for python libraries like torch, matplotlib, numpy, etc... as this repository doesn't include them directly but the user has to download them (via pip / conda install ...)
But when distributing a compiled pre-pass .wheel, the parts of the dependencies are included in the binaries. Not as source code, but compiled, which still seems to require to include the Apache license. (Not too sure about MIT, doesn't read like it but I'm not a lawyer).

### tl;dr:
We should include the Licenses and copyright-notices for all dependencies of the pre-pass in a pre-pass/licenses folder, just to be on the save side.

### `cargo license` output:
**(MIT OR Apache-2.0) AND Unicode-DFS-2016** (1): unicode-ident
**Apache-2.0** (5): pyo3, pyo3-build-config, pyo3-ffi, pyo3-macros, pyo3-macros-backend
**Apache-2.0 OR MIT** (23): autocfg, bitflags, cfg-if, indoc, libc, lock_api, once_cell, parking_lot, parking_lot_core, proc-macro2, quote, scopeguard, smallvec, syn, unindent, windows-sys, windows_aarch64_gnullvm, windows_aarch64_msvc, windows_i686_gnu, windows_i686_msvc, windows_x86_64_gnu, windows_x86_64_gnullvm, windows_x86_64_msvc
**Apache-2.0 WITH LLVM-exception** (1): target-lexicon
**MIT** (1): redox_syscall

**Comment by mzaiss** [comment]
just add it for now.
maybe we can create a non-compiled prepass at some point and provide the compiled version upon request.

**Comment by J-Endres** [comment]
Added a file containing license info from above https://github.com/mzaiss/MRTwin_pulseq/commit/2557f5268214ab72a50d9e852b7448c704ddc83d
I didn't include their license files as MIT contains the name of the copyright owner and I would have to include the licenses separately for every dependency. Instead, I referenced where to find them.

Note that it seems like Apache is not fully compatible with (A)GPL which we use. It seems like one must be allowed to still use the apache-licensed part under apache instead of AGPL, some places say that one can assume that we implicitely allow this.
So it doesn't seem like a serious problem, but it we may look into that at some point.
